contracts/LiquidityBridgeContractV2.sol

```diff
@@ -255,28 +255,39 @@ contract LiquidityBridgeContractV2 is Initializable, OwnableUpgradeable, Reentra
         return (providerId);
     }
 
-    function getProviders(
-        uint[] memory providerIds
-    ) external view returns (LiquidityProvider[] memory) {
-        LiquidityProvider[] memory providersToReturn = new LiquidityProvider[](
-            providerIds.length
-        );
+    function getProviders() external view returns (LiquidityProvider[] memory) {
         uint count = 0;
-
-        for (uint i = 0; i < providerIds.length; i++) {
-            uint id = providerIds[i];
-            if (
-                (isRegistered(liquidityProviders[id].provider) ||
-                    isRegisteredForPegout(liquidityProviders[id].provider)) &&
-                liquidityProviders[id].status
-            ) {
-                providersToReturn[count] = liquidityProviders[id];
+        LiquidityProvider storage lp;
+        for (uint i = 1; i <= providerId; i++) {
+            if (shouldBeListed(liquidityProviders[i])) {
```

Comment:
Help me please to understand this method

Reply:
Not sure if you asked about `shouldBeListed` or `getProviders` but I'll explain both:
- `shouldBeListed`: returns true if the LP should be included in the list of available LPs, this means is registered for pegin or pegout (meaning that has enough locked collateral and hasn't resigned) and has enabled status
- `getProviders`: this function returns the array of available LPs, since the `push` method is not available on memory arrays, we need to know the size of the list first, therefore, first we count how many providers accomplish the condition to be enabled and then we create the array and populate it

Reply:
Noted, i was referring to the case with lp enforcement initiative which should return the list for a specific scope/partner

Reply:
ahhh, about LP enforcement initiative, IMO it should be handle off chain, we can discuss further once we take the epic into design phase